### PR TITLE
Add Supabase Authentication

### PR DESCRIPTION
**Pull request by MrF3lix**
- Add Welcome Page
- Add Login Page
- Add Register Page
- Add Magic Link Page
- Allow google login
- Allow apple login

### CODE REVIEW

**Comment by vkaelin** [comment]
Nice, looks good! 

Shouldn't we add `.env` file in `.gitignore` and add another one `.env.example` for example?

So we can just `cp .env.example .env` when we setup the repo for the first time.

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
+        path === '/email-login'
+      )
+    ) {
+      navigate('/welcome');
+    }
+  }, [session, location, setSession, navigate]);
+
+  return (
+    <Suspense fallback={<p>Loading...</p>}>
+      <div className="flex h-[100vh] w-[100vw] bg-gray-50">
```
Comment (vkaelin) [inline src/App.tsx]:
```suggestion
      <div className="flex h-screen w-screen bg-gray-50">
```

**Comment by MrF3lix** [comment]
@vkaelin Regarding the .env file. In past projects we've always used the .env file as the example and the non secret configuration and a .env.local file with the secrets.

**Comment by vkaelin** [comment]
> @vkaelin Regarding the .env file. In past projects we've always used the .env file as the example and the non secret configuration and a .env.local file with the secrets.

Okok let's try that! Maybe just add a comment about how to configure your environnement in the README. And just add in the `.gitignore` file the other env files.

**Review by vkaelin — APPROVED** [review]

**Comment by vkaelin** [comment]
Merging it so I can work more easily on the app. 

Feel free to open a new PR if you still have issues to resolve with the auth.

**Pull request merged**